components/Navigator/components/SearchButton/index.jsx

```diff
@@ -2,19 +2,17 @@ import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 import MagnifierIcon from '@shopgate/pwa-ui-shared/icons/MagnifierIcon';
 import Ripple from '@shopgate/pwa-ui-shared/Ripple';
+import NavigatorContext from '../../context';
 import styles from './style';
 
 /**
  * The search button component.
  */
 class SearchButton extends Component {
```

Comment:
You should also make this a `PureComponent`. You only receive primitive props.

Reply:
In this case, there is an explicit check inside stopping the component from updating.